client/src/hooks.ts

```diff
@@ -298,6 +301,7 @@ export function useViewed(callback: Function) {
           timer.current = {
             timeout: window.setTimeout(() => {
               timer.current = { timeout: -1 };
+              setNode(undefined);
```

Comment:
This resolves a performance issue, because the IntersectionObserver was constantly firing when scrolling over the BCD table.

Reply:
I feel like it would be better to have a separate `useState()` to essentially disconnect the `IntersectionObserver`. It could be `const [isViewed, setViewed] = useState(false);` and we could have a computed value `const observedNode = useMemo(() => isViewed ? undefined : node, [isViewed, node]);` and pass `observedNode` to `useIsIntersecting()`. Wdyt? @fiji-flo

Reply:
```suggestion
```

We please keep it as is. While yes we can do `setNode(undefined)` here,  let's please do this in a separate PR (ideally not now).

> This resolves a performance issue, because the IntersectionObserver was constantly firing when scrolling over the BCD table.

Can you elaborate on this. As far as I know the intersection observer fires only when entering or leaving with a threshold `[0]`. So if you mean that with constantly, that's not an performance issue. I think our dev-server does fancy reloads and this might now play well together.

Reply:
Extracted to https://github.com/mdn/yari/pull/12067 without the (unnecessary) `setNode(undefined)` call.